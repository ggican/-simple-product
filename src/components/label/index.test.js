import React from "react";
import Label from "./index";
import StyleLabel from "./index.style";

describe("Label Component", () => {
    let result;
    const onCloseLabel = value => {
        result = value;
    };
    let successElement;
    let failedElement;

    beforeEach(() => {
        successElement = (
            <Label onCloseLabel={onCloseLabel} title="Cooking"></Label>
        );
        failedElement = <Label></Label>;
    });

    it("SUCCESS renders and create snapshot  <Label />", () => {
        expect(toJson(mount(successElement))).toMatchSnapshot();
    });

    it("SUCCESS renders style  <Label />", () => {
        expect(toJson(mount(<StyleLabel></StyleLabel>))).toMatchSnapshot();
    });

    it("SUCCESS renders succe  <Label />", () => {
        expect(
            shallow(successElement)
                .find(".label__text")
                .text(),
        ).toEqual("Cooking");
    });

    it("SUCCESS renders OnClickCloseLabel  <Label />", () => {
        shallow(successElement)
            .find(".label__close")
            .simulate("click", {
                preventDefault: () => {},
            });
        expect(result).toEqual("Cooking");
    });
    it("FAILED renders Went Does't have title  <Label />", () => {
        expect(
            shallow(failedElement)
                .find(".label__text")
                .text(),
        ).toEqual("-");
    });
});
