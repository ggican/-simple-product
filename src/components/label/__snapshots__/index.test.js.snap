// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Component SUCCESS renders and create snapshot  <Label /> 1`] = `
<Label
  onCloseLabel={[Function]}
  title="Cooking"
>
  <styled.div>
    <div
      className="sc-AxjAm NVobY"
    >
      <div
        className="label"
      >
        <div
          className="label__text"
        >
          Cooking
        </div>
        <button
          className="label__close"
          onClick={[Function]}
          type="button"
        >
          X
        </button>
      </div>
    </div>
  </styled.div>
</Label>
`;

exports[`Label Component SUCCESS renders style  <Label /> 1`] = `
<styled.div>
  <div
    className="sc-AxjAm NVobY"
  />
</styled.div>
`;
